Modules/Loadable/Segmentations/EditorEffects/qSlicerSegmentEditorAbstractEffect.cxx

```diff
@@ -300,7 +300,10 @@ void qSlicerSegmentEditorAbstractEffect::modifySegmentByLabelmap(vtkMRMLSegmenta
   // Prevent disappearing and reappearing of 3D representation during update
   SlicerRenderBlocker renderBlocker;
 
-  vtkSmartPointer<vtkOrientedImageData> modifierLabelmap = modifierLabelmapInput;
+  // Prevent the original modifier labelmap from being modified
+  vtkNew<vtkOrientedImageData> modifierLabelmap;
```

Comment:
Copying the modifier labelmap could might require significant extra memory and maybe even cause perceivable amount of delay if the the modifier labelmap is large. Could we make the copy only when it is necessary (e.g., mask is the same as the selected segment)?

Reply:
I think that a deep copy is necessary any time that we want to use any sort of masking, as the mask is applied directly to the modifier labelmap. The alternatives would be to document that the input modifier labelmap can be changed by the function, or to change the way that masking works to that the modifier labelmap is not changed. In any case, I've moved the DeepCopy call so that it won't be called if no masking options are enabled.

Reply:
> document that the input modifier labelmap can be changed by the function

This would be OK, but probably developers would not expect it and it might cause problems.

If the modifier is only copied when masking is enabled then I think that should be good enough (there is some overhead due to masking anyway, so one extra buffer and a copy would probably not be noticeable).